src/generators/javascript/presets/CommonPreset.ts

```diff
@@ -0,0 +1,213 @@
+import { JavaScriptRenderer } from '../JavaScriptRenderer';
+import { JavaScriptPreset } from '../JavaScriptPreset';
+import { getUniquePropertyName, DefaultPropertyNames, TypeHelpers, ModelKind } from '../../../helpers';
+import { CommonInputModel, CommonModel } from '../../../models';
+import renderExampleFunction from './utils/ExampleFunction';
+
+export interface JavaScriptCommonPresetOptions {
+  marshalling: boolean;
+  example: boolean;
```

Comment:
For this PR it would probably make sense to only include marshalling functionality 🙂 Feel free to create another with the examples feature if you see it useful 🙂

Reply:
Yeah, actually I am confused with this part implementation because the whole code is written in typescript.

Reply:
The implementation of Modelina and the presets are written in TypeScript and should stay that way. It is only the generated code that needs to be converted to JS.